bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/NotificationAction.java

```diff
@@ -57,9 +58,31 @@ public static void sendNotification(String userId, String message) {
     @ActionDoc(text = "Sends a push notification to mobile devices of user with userId")
     public static void sendNotification(String userId, String message, @Nullable String icon,
             @Nullable String severity) {
+        sendNotification(userId, message, icon, severity, null, null, null, null, null);
+    }
+
+    /**
+     * Sends an advanced push notification to mobile devices of user
+     *
+     * @param userId the cloud user id of the recipient
+     * @param message the body of the notification
+     * @param icon name for the notification
+     * @param severity category for the notification
+     * @param onClickAction the action to perform when clicked
+     * @param mediaAttachmentUrl the media to attach to a notification
+     * @param actionButton1 an action button in the format "Title=Action"
+     * @param actionButton2 an action button in the format "Title=Action"
+     * @param actionButton3 an action button in the format "Title=Action"
```

Comment:
I guess you don't want to make this a `List<String>` to outline the number of actions being limited? I think I'd prefer doing that if there's no strong reason against it, and print some warning message if the list size exceeds 3.

Reply:
What IMO speaks against this, is that constructing a `List<String>` is more complicated for script languages than simply passing three string args.

Reply:
That applies to script languages with helper libraries, but not to rules DSL users. I'm not sure about how many there are, but forum questions suggest it's still a sizable portion of our user base?

Reply:
After looking around the code, my assumption was that only simple primitive types would work across rules and actions ( I can't remember where in the core, but i stumbled across a class that defined these) , hence why not using more complex structures.